projects/smoke_mcp2515/src/main.c

```diff
@@ -0,0 +1,83 @@
+// Simple smoketest project for MCP2515 boards.
+
+// Periodically transmits a message over the MCP2515, and logs any
+// messages recieved over the MCP2515
+
+#include "can.h"
+#include "can_msg_defs.h"
+#include "delay.h"
+#include "event_queue.h"
+#include "generic_can.h"
+#include "generic_can_mcp2515.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "mcp2515.h"
+#include "soft_timer.h"
+
+#define TEST_DLC 8
+#define TEST_EXTENDED false
+
+static SystemCanDevice s_id = 123;
+
+static GenericCanMcp2515 s_can_mcp2515;
+
+static void prv_rx_callback(const GenericCanMsg *msg, void *context) {
+  LOG_DEBUG("Received a message!\n");
+  char log_message[30];
+  printf("Data:\n\t");
+  uint8_t i;
+  for (i = 0; i < msg->dlc; i++) {
+    uint8_t byte = 0;
+    byte = msg->data >> (i * 8);
+    printf("%x ", byte);
```

Comment:
Is there a reason we're using `printf` rather than `LOG_DEBUG`?

Reply:
nope, changed it.